LiteralColorExpression tests

ggetz opened the pull request:
* Renamed `ConstantColorExpression` to `LiteralColorExpression`
* Wrote Spec for `LiteralColorExpression`

pjcozzi commented on Source/Scene/Cesium3DTileStyle.js:
```diff
@@ -8,7 +8,7 @@ define([
        './ColorRampExpression',
        './ColorMapExpression',
-       './ConstantColorExpression',
+       './LiteralColorExpression',
```
Search and replace throughout this file (and then test in Sandcastle to double check).

pjcozzi commented on Specs/Scene/LiteralColorExpressionSpec.js:
```diff
@@ -0,0 +1,64 @@
+    };
+
+    it('constructs with hex', function() {
+        var expression = new LiteralColorExpression(new MockStyleEngine(), "#ffffff");
```
Use single quotes for all the strings in this file.  We usually always use single except for JSON examples, e.g., the CZML examples in Sandcastle, and the style examples we have been making in Sandcastle.

pjcozzi commented:
Check for the `DeveloperError`s.  For example, https://github.com/ggetz/cesium/blob/literal-color-exp-tests/Specs/Core/Cartesian2Spec.js#L42-L46

pjcozzi commented:
>          if (color === undefined) {
             throw new DeveloperError('color must be defined');
         }

Use `!defined` (same comment throughout).

Also, looking more carefully at this, the string should be "color must be a valid CSS string" despite me saying something else previously.  Sorry.

pjcozzi commented:
Nice work!  Just those comments.

ggetz commented:
I made the updates and added the test to check for the `DeveloperError`. Also I removed the DeveloperError in the setter for `color` because it sets the color directly, not by constructing it with a css string.

pjcozzi commented:
Thanks!
